### ISSUE TITLE
Custom SwiftUI component untapable

### ISSUE THREAD

**Issue by callaars** [wix/react-native-navigation#6678]
### Issue Description

I've got a custom SwiftUI component with a `Button`. That button, however, does not seem to recognise the tap. I know it's due to how RNN uses the `rootViewController` but I can't seem to figure out how to mitigate this.

I'm sorry if this is not the right place for it, but I can't seem to get answers anywhere else.

Are there any known workarounds for this?

### Steps to Reproduce / Code Snippets / Screenshots

**ReorderableList.swift**
```swift
import SwiftUI

struct ReorderableList : View {  
  var body: some View {
    VStack {
      Button(action: {
        print("Button tapped")
      }) {
        Image(systemName: "plus.circle.fill")
          .foregroundColor(.white)
          .padding()
          .background(Color.red)
          .clipShape(Circle())
      }
    }
    .padding()
    .clipShape(Circle())
  }
}
```

**ReorderableListManager.m**
```objc
#import <Foundation/Foundation.h>
#import "React/RCTViewManager.h"
#import "React/RCTComponentEvent.h"
#import "TimeToBake-Swift.h"

@interface ReorderableListManager : RCTViewManager
@end

@implementation ReorderableListManager

RCT_EXPORT_MODULE()

- (UIView *)view {
  ReorderableListProxy *proxy = [[ReorderableListProxy alloc] init];
  UIView *view = [proxy view];
  NSMutableDictionary *storage = [ReorderableListProxy storage];
  storage[[NSValue valueWithNonretainedObject:view]] = proxy;
  return view;
}

@end
```

**ReorderableProxy.swift**
```swift
import SwiftUI

@objcMembers class ReorderableListProxy: NSObject {
  private var vc: UIHostingController<ReorderableList>
  
  static let storage = NSMutableDictionary()
  
  var count: Int {
    set { vc.rootView.props.count = newValue }
    get { return vc.rootView.props.count }
  }
  
  var onCountChange: RCTBubblingEventBlock {
    set { vc.rootView.props.onCountChange = newValue }
    get { return vc.rootView.props.onCountChange }
  }
  
  var view: UIView {
    return vc.view
  }
  
  override init() {
    vc = UIHostingController(rootView: ReorderableList())
  }
}
```

**some-file.tsx**
```tsx
import React from 'react';
import {requireNativeComponent} from 'react-native';

const ReorderableList = requireNativeComponent('ReorderableList');
const List = () => (
  <ReorderableList />
);

export default List;
```
---
### Environment
* React Native Navigation version: 7.0.0
* React Native version: 0.63.2
* Platform(s) (iOS, Android, or both?): iOS
* Device info (Simulator/Device? OS version? Debug/Release?): Simulator, debug

**Comment by danilobuerger** [comment]
I don't think you will get much of a response about it here. I think your best bet would be to do a deep dive yourself and fix it.

**Comment by danilobuerger** [comment]
One thing I noticed, I don't think your usage of the `UIHostingController` is correct. This should be embedded, as per the docs: 

> Use the hosting controller like you would any other view controller, by presenting it or embedding it as a child view controller in your interface.

You are instead just returning its containing view which might work in some cases (showing up on screen) and not others (intercepting touches). Maybe.

**Comment by callaars** [comment]
Thanks @danilobuerger I didn't even see that. You're right.

**Comment by danilobuerger** [comment]
So I got curious and investigated it. It works without a problem for me, after specifying a width and a height in js land:

SwiftUIButtonProxy.swift
```swift
@objcMembers class SwiftUIButtonProxy: NSObject {
    @objc var vc: UIViewController = UIHostingController(rootView: SwiftUIButton())
}
```

SwiftUIButtonManager.m
```objc
@implementation SwiftUIButtonManager

RCT_EXPORT_MODULE(SwiftUIButtonManager);

-(UIView *)view {
    if (@available(iOS 13.0, *)) {
        SwiftUIButtonProxy *proxy = [[SwiftUIButtonProxy alloc] init];
        return proxy.vc.view;
    }
    return nil;
}

@end
```

Button.tsx
```tsx
const SwiftUIButton = requireNativeComponent("SwiftUIButton");
const Button = () => (
  <SwiftUIButton style={{ width: 100, height: 100 }} />
);
```

**Comment by callaars** [comment]
Hm, I'll check when I'm home. You're the real MVP.
